cmds/core/chmod/chmod.go

```diff
@@ -21,45 +22,47 @@ import (
 	"regexp"
 	"strconv"
 	"strings"
-
-	"github.com/u-root/u-root/pkg/uroot/util"
 )
 
-const special = 99999
-
-var (
-	recursive = flag.Bool("recursive", false, "do changes recursively")
-	reference = flag.String("reference", "", "use mode from reference file")
+const (
+	special = 99999
+	usage   = "chmod: chmod [mode] filepath"
 )
 
-var usage = "chmod: chmod [mode] filepath"
+var errBadUsage = errors.New(usage)
 
 func init() {
-	util.Usage(usage)
+
 }
 
 func changeMode(path string, mode os.FileMode, octval uint64, mask uint64) (fs.FileMode, error) {
 	// A special value for mask means the mode is fully described
 	if mask == special {
-		return mode, os.Chmod(path, mode)
+		if err := os.Chmod(path, mode); err != nil {
+			return 0, err
+		}
+		return mode, nil
 	}
 
 	var info os.FileInfo
 	info, err := os.Stat(path)
 	if err != nil {
-		return mode, err
+		return 0, err
 	}
 	mode = info.Mode() & os.FileMode(mask)
 	mode = mode | os.FileMode(octval)
 
-	return mode, os.Chmod(path, mode)
+	if err := os.Chmod(path, mode); err != nil {
+		return 0, err
+	}
+	return mode, nil
 }
 
 func calculateMode(modeString string) (mode os.FileMode, octval uint64, mask uint64, err error) {
 	octval, err = strconv.ParseUint(modeString, 8, 32)
 	if err == nil {
 		if octval > 0o777 {
-			return mode, octval, mask, fmt.Errorf("invalid octal value %0o. Value should be less than or equal to 0777", octval)
+			return mode, octval, mask, fmt.Errorf("%w: invalid octal value %0o. Value should be less than or equal to 0777", strconv.ErrRange, octval)
```

Comment:
```suggestion
			return mode, octval, mask, fmt.Errorf("invalid octal value %0o. Value should be less than or equal to 0777: %w", octval, strconv.ErrRange)
```

I typically see the %w appear at the end of the error string. For example in https://go.dev/blog/go1.13-errors

Reply:
I agree but in this case, where I'm picking a specific error, the text reads better and it works anyway?